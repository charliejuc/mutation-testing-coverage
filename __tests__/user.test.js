// user.test.js
const User = require("../src/user");

test("getUserInfo returns correct object for adult", () => {
  const user = new User("John Doe", 30);
  expect(user.getUserInfo()).toMatchSnapshot();
});

test("getUserInfo with different user", () => {
  const user = new User("Jane Doe", 25);
  expect(user.getUserInfo()).toMatchSnapshot();
});

test("getUserInfo returns correct object for minor", () => {
  const user = new User("Johnny Doe", 16);
  expect(user.getUserInfo()).toMatchSnapshot();
});

test("isAdult returns true for age >= 18", () => {
  const user = new User("John Doe", 30);
  expect(user.isAdult()).toBe(true);
});
