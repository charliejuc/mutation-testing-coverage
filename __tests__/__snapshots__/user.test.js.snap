// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUserInfo returns correct object for adult 1`] = `
{
  "age": 30,
  "description": "Adult",
  "name": "John Doe",
}
`;

exports[`getUserInfo returns correct object for minor 1`] = `
{
  "age": 16,
  "description": "Young",
  "name": "Johnny Doe",
}
`;

exports[`getUserInfo with different user 1`] = `
{
  "age": 25,
  "description": "Adult",
  "name": "Jane Doe",
}
`;
